actors/builtin/market/market_actor.go

```diff
@@ -505,25 +512,20 @@ func (a Actor) CronTick(rt Runtime, _ *adt.EmptyValue) *adt.EmptyValue {
 				}
 
 				slashAmount, nextEpoch, removeDeal := msm.updatePendingDealState(rt, state, deal, dealID, rt.CurrEpoch())
+				Assert(slashAmount.GreaterThanEqual(big.Zero()))
 				if removeDeal {
-					if err := deleteDealProposalAndState(dealID, msm.dealStates, msm.dealProposals, true, true); err != nil {
-						builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to delete deal")
-					}
-				}
-				if !slashAmount.IsZero() {
-					amountSlashed = big.Add(amountSlashed, slashAmount)
-				}
+					Assert(nextEpoch == epochUndefined)
 
-				if nextEpoch != epochUndefined {
-					Assert(nextEpoch > rt.CurrEpoch())
+					amountSlashed = big.Add(amountSlashed, slashAmount)
+					err := deleteDealProposalAndState(dealID, msm.dealStates, msm.dealProposals, true, true)
+					builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to delete deal proposal and states")
+				} else {
+					Assert(nextEpoch > rt.CurrEpoch() && amountSlashed.IsZero())
```

Comment:
Oops, I think my suggestion was wrong on this one. It's not `amountSlashed.IsZero()`, it's `slashAmount.IsZero()`.